app/services/ipam/dhcp.rb

```diff
@@ -2,7 +2,10 @@ module IPAM
   class Dhcp < Base
     delegate :dhcp, :dhcp_proxy, :to => :subnet
     def suggest_ip
-      return unless subnet.dhcp?
+      unless subnet.dhcp?
+        logger.info "Unable to suggest IP: no DHCP proxy or IPAM set for subnet '#{subnet.name}' or taxonomy does not match host"
+        return
```

Comment:
Should we also add an error here (like we do below) so the user sees this in the UI and not just the logs?

Reply:
Very good idea actually! I am also improving code here and below so we DRY.